crates/configuration/src/version4/to_runtime_configuration.rs

```diff
@@ -73,16 +76,35 @@ fn convert_scalar_types(
                             .into_iter()
                             .map(|(k, v)| (k, convert_comparison_operator(v)))
                             .collect(),
-                        type_representation: scalar_type
-                            .type_representation
-                            .map(convert_type_representation),
+                        type_representation: convert_or_infer_type_representation(
+                            scalar_type.type_representation,
+                            &scalar_type_name,
+                        ),
                     },
                 )
             })
             .collect(),
     )
 }
 
+/// Infer scalar type representation from scalar type name, if necessary. Defaults to JSON representation
+fn convert_or_infer_type_representation(
```

Comment:
V0.2.0 requires type representation.

Type representation comes from introspection configuration, and may be absent.

So, if type representation is missing, we infer the type based on the name and fetch the corresponding type representation from the default introspection configuration.